modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/configuration/admin/service/OpenIdConnectProviderManagedServiceFactory.java

```diff
@@ -475,31 +491,41 @@ private void _updateOAuthClientEntry(
 		}
 
 		try {
-			String authServerWellKnownURI = _updateOAuthClientASLocalMetadata(
-				defaultUserId, properties);
-
-			OAuthClientEntry oAuthClientEntry =
-				_oAuthClientEntryLocalService.fetchOAuthClientEntry(
-					companyId, authServerWellKnownURI,
-					_getPropertyAsString("openIdConnectClientId", properties));
-
-			if (oAuthClientEntry == null) {
-				oAuthClientEntry =
-					_oAuthClientEntryLocalService.addOAuthClientEntry(
-						defaultUserId,
-						_generateAuthRequestParametersJSON(properties),
-						authServerWellKnownURI, _generateInfoJSON(properties),
-						OAuthClientEntryConstants.OIDC_USER_INFO_MAPPER_JSON,
-						_generateTokenRequestParametersJSON(properties));
+			OAuthClientEntry oAuthClientEntry = null;
+
+			if (oldProperties != null) {
+				String oldAuthServerWellKnownURI =
+					_updateOAuthClientASLocalMetadata(
+						defaultUserId, oldProperties);
+
+				OAuthClientEntry oldOAuthClientEntry =
+					_oAuthClientEntryLocalService.fetchOAuthClientEntry(
+						companyId, oldAuthServerWellKnownURI,
+						_getPropertyAsString(
+							"openIdConnectClientId", oldProperties));
+
+				if (oldOAuthClientEntry != null) {
+					String authServerWellKnownURI =
+						_updateOAuthClientASLocalMetadata(
+							defaultUserId, properties);
+
+					oAuthClientEntry =
+						_oAuthClientEntryLocalService.updateOAuthClientEntry(
+							oldOAuthClientEntry.getOAuthClientEntryId(),
+							_generateAuthRequestParametersJSON(properties),
+							authServerWellKnownURI,
+							_generateInfoJSON(properties),
+							oldOAuthClientEntry.getOIDCUserInfoMapperJSON(),
+							_generateTokenRequestParametersJSON(properties));
+				}
+				else {
+					oAuthClientEntry = _addOAuthClientEntry(
+						properties, defaultUserId);
```

Comment:
If we create a "Provider Connection" with `null` "OpenID Connect Client ID" and then update the entry with valid "OpenID Connect Client ID", the `oldProperties` is not `null` but `oldOAuthClientEntry` is null. A new `OAuthClientEntry` entry would be created.